### DISCUSSION TITLE
GLB model texture shows as white in IOS Quick Look

### DISCUSSION THREAD

**Discussion by omnia-tom** [google/model-viewer#4326] [Q&A]
### Description
I am currently having an issue when showing my GLB file in IOS AR quick view. All the models are coming in as white textured. This is not expected, because in all other GLB viewers, the textures come through. It also shows correctly in the normal model-viewer, as seen in the video posted below.

I figured this might be due to the conversion process between GLB and USDZ for IOS. According to the documentation, it states:

_Note that by not specifying an ios-src, the USDZ will instead be auto-generated when the user clicks the AR button on iOS to launch Quick Look._

Is there any additional settings I need to enable textures?  
This is the current way I am setting things up:

```const { modelname } = props?.params;
useEffect(() => {
    if ('xr' in navigator && navigator.xr.isXRSupported) {
      navigator.xr.requestSession('immersive-ar').then((session) => {
        const viewer = modelViewerRef.current;
        if (viewer && viewer.canActivateAR) {
          viewer.canActivateAR().then(() => {
            viewer.activateAR(session);
          });
        }
      });
}
...
<model-viewer
      ref={modelViewerRef}
      src={my-model}
      ar
      camera-controls
      orientation="0deg -90deg 0deg"
      style={{ width: '100%', height: '500px' }}
    >
</model-viewer>
```
    

#### Live Demo

https://github.com/google/model-viewer/assets/129299233/d3f52092-26e3-4f5f-90d5-f03e4472ee7f

### Version

- model-viewer: v3.0.1

### Browser Affected

- [ ] Chrome, version: xx.x.xxxx.xx
- [ ] Edge
- [ ] Firefox
- [ ] IE
- [X] Safari

### OS

- [ ] Android
- [X] iOS
- [ ] Linux
- [ ] MacOS
- [ ] Windows

### AR

- [ ] WebXR
- [ ] SceneViewer
- [X] QuickLook

**Comment by elalish** [comment]
Our auto-generation is best-effort and does not support everything. If it fails, you should use Apple's Reality Converter and supply an `ios-src`. In order for anyone to help, you'll need to supply your problem GLB file. However, since we use the three.js USDZExporter for this feature, you'd probably be better off filing the bug on three.js directly.

**Reply by elalish** [reply]
Speaking of glTF->USDZ conversion problems, here's one for you, @meshula.

**Reply by meshula** [reply]
Thanks Emmett.

cc @jesschimein
